kernel/net/src/interface.rs

```diff
@@ -56,14 +56,14 @@ impl NetworkInterface {
     }
 
     /// Adds a socket to the interface.
-    pub fn add_socket<T>(&self, socket: T) -> Socket<T>
+    pub fn add_socket<T>(self: &Arc<Self>, socket: T) -> Socket<T>
```

Comment:
if you always clone it, it's likely better to just accept `Arc<Self>` instead of a reference to make it clearer how the internal socket ownership is handled.

Reply:
Yea that's something I initially did, but switched it because cloning an interface to add a socket didn't really make sense. I'll change it back because this is something I hope to eventually change anyway when I update the `net` API.